standard/PortalStatistics.lua

```diff
@@ -0,0 +1,1138 @@
+				data[opponent][placement] = data[opponent][placement] + 1
+			end
+		end
+	end
+
+	Lpdb.executeMassQuery('placement', queryParameters, processData)
+
+	return data
+end
+
+
+--[[
+Section: Display Functions
+]]--
+
+
+function StatisticsPortal._earningsTableHeader(args)
+
+	local columnText = args.opponentType == Opponent.team and 'Organization' or 'Player'
+
+	local row = mw.html.create('tr')
+		:tag('th'):wikitext('#'):addClass('unsortable')
+		:tag('th'):wikitext(columnText):addClass('unsortable')
+		:tag('th'):wikitext('Achievements'):css('width', '200px'):addClass('unsortable')
+		:tag('th'):wikitext(Medal['1'])
+		:tag('th'):wikitext(Medal['2'])
+		:tag('th'):wikitext(Medal['3'])
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('th'):wikitext('Show<br>Match')
+	end
+
+	row:tag('th')
+		:tag('abbr')
+		:attr('title', 'Total earnings across all games')
+		:wikitext('Earnings')
+
+	return row:allDone()
+end
+
+
+function StatisticsPortal._earningsTableRow(args, placements, earnings, opponentIndex, opponentDisplay)
+
+	local row = mw.html.create('tr')
+		:css('line-height', '25px')
+		:css('text-align', 'center')
+		:tag('td'):wikitext(opponentIndex)
+		:tag('td'):css('text-align', 'left'):node(opponentDisplay)
+		:tag('td'):wikitext(StatisticsPortal._achievementsDisplay(placements['sWinData'] or {}))
+		:tag('td'):wikitext(placements['1'] or '0')
+		:tag('td'):wikitext(placements['2'] or '0')
+		:tag('td'):wikitext(placements['3'] or '0')
+
+	if Logic.readBool(args.displayShowMatches) then
+		row:tag('td'):wikitext(placements['showWins'] or '0')
+	end
+
+	row:tag('td')
+		:css('text-align', 'right')
+		:wikitext('$' .. Currency.formatMoney(earnings))
+
+	return row:allDone()
+end
+
+
+function StatisticsPortal._achievementsDisplay(data)
+	local output = ''
+	if data and type(data[1]) == 'table' then
+		for _, item in ipairs(data) do
+			item.icon = string.gsub(item.icon, 'File:', '')
+			item.icondark = string.gsub(item.icondark or '', 'File:', '')
+			item.icon = String.isNotEmpty(item.icon) and item.icon or 'Gold.png'  --'InfoboxIcon Tournament.png'
+			output = output .. LeagueIcon.display{
+				icon = item.icon,
+				iconDark = item.icondark,
+				link = item.pagename,
+				name = item.shortname,
+				options = { noTemplate = true },
+			}
+			output = output .. ' '
+		end
+	end
+	return output
+end
+
+
+function StatisticsPortal._drawChart(config, chartData)
+	return mw.html.create('div'):node(mw.ext.Charts.chart({
+			size = {
+				height = config.height,
+				width = config.width,
+			},
+			tooltip = {
+				trigger = 'axis',
+			},
+			legend = config.customLegend,
+			yAxis = config.flipAxes and config.xAxis or config.yAxis,
+			xAxis = config.flipAxes and config.yAxis or config.xAxis,
+			series = chartData,
+			labels = config.labels,
+		})
+	)
+end
+
+
+--[[
+Section: Utility Functions
+]]--
+
+
+function StatisticsPortal._buildChartData(config, yearSeriesData, nonYearCategories, transpose)
```

Comment:
if i see it correctly on phone this function does more that build the data

Reply:
It takes the raw data from the querying, alongside some config options, and then formats it to be passed to the chart drawing function

Reply:
Earlier comment saying this is too big, I have moved the other part to a function, hopefully making the function of a good size